app/src/main/java/org/fossasia/pslab/others/ZoomLayout.java

```diff
@@ -0,0 +1,132 @@
+    }
+
+    public ZoomLayout(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        init(context);
+    }
+
+    public ZoomLayout(Context context, AttributeSet attrs, int defStyle) {
+        super(context, attrs, defStyle);
+        init(context);
+    }
+
+    @SuppressLint("ClickableViewAccessibility")
+    public void init(Context context) {
+        final ScaleGestureDetector scaleDetector = new ScaleGestureDetector(context, this);
+        this.setOnTouchListener(new OnTouchListener() {
+            @Override
+            public boolean onTouch(View view, MotionEvent motionEvent) {
+                switch (motionEvent.getAction() & MotionEvent.ACTION_MASK) {
+                    case MotionEvent.ACTION_DOWN:
+                        if (scale > MIN_ZOOM) {
+                            mode = Mode.DRAG;
+                            startX = motionEvent.getX() - prevDx;
+                            startY = motionEvent.getY() - prevDy;
+                        }
+                        break;
+                    case MotionEvent.ACTION_MOVE:
+                        if (mode == Mode.DRAG) {
+                            dx = motionEvent.getX() - startX;
+                            dy = motionEvent.getY() - startY;
+                        }
+                        break;
+                    case MotionEvent.ACTION_POINTER_DOWN:
+                        mode = Mode.ZOOM;
+                        break;
+                    case MotionEvent.ACTION_POINTER_UP:
+                        mode = Mode.DRAG;
+                        break;
+                    case MotionEvent.ACTION_UP:
+                        mode = Mode.NONE;
+                        prevDx = dx;
+                        prevDy = dy;
+                        break;
+                }
+                scaleDetector.onTouchEvent(motionEvent);
+
+                if ((mode == Mode.DRAG && scale >= MIN_ZOOM) || mode == Mode.ZOOM) {
+                    getParent().requestDisallowInterceptTouchEvent(true);
+                    float maxDx = (child().getWidth() - (child().getWidth() / scale)) / 2 * scale;
+                    float maxDy = (child().getHeight() - (child().getHeight() / scale)) / 2 * scale;
+                    dx = Math.min(Math.max(dx, -maxDx), maxDx);
+                    dy = Math.min(Math.max(dy, -maxDy), maxDy);
+                    Log.i(TAG, "Width: " + child().getWidth() + ", scale " + scale + ", dx " + dx
+                            + ", max " + maxDx);
+                    applyScaleAndTranslation();
+                }
+                return true;
+            }
+        });
+    }
+
+    @Override
+    public boolean onScaleBegin(ScaleGestureDetector scaleDetector) {
+        return true;
+    }
+
+    // ScaleGestureDetector
+    @Override
+    public boolean onScale(ScaleGestureDetector scaleDetector) {
+        float scaleFactor = scaleDetector.getScaleFactor();
+        Log.i(TAG, "onScale " + scaleFactor);
```

Comment:
this too

Reply:
It shows the scale factor i.e. how much the screen is zoomed.